qiskit_nature/problems/sampling/protein_folding/peptide/chains/main_chain.py

```diff
@@ -26,77 +26,64 @@ class MainChain(BaseChain):
+            main_chain_residue_sequence: String of characters that define residues for the main
                                         chain.
-            side_chain_lens: List of lengths of all side chains.
             side_chain_residue_sequences: List of characters that define residues for all side
-                                        beads. None if a side bead does not exist.
+                                        beads. Empty string if a side bead does not exist.
 
         Raises:
-            InvalidSizeException: when the length of list of side chain lengths provided does not
+            InvalidSizeException: If the length of list of side chain lengths provided does not
                                     equal the length of the main chain.
-            InvalidSideChainException: when first, second or last main beads have a side chain.
+            InvalidSideChainException: If first, second or last main beads have a side chain.
         """
         self._main_chain_residue_sequence = main_chain_residue_sequence
         beads_list = self._build_main_chain(
-            main_chain_len,
             main_chain_residue_sequence,
-            side_chain_lens,
             side_chain_residue_sequences,
         )
         super().__init__(beads_list)
 
     @property
-    def main_chain_residue_sequence(self) -> List[str]:
+    def main_chain_residue_sequence(self) -> str:
         """Returns a residue sequence for the main chain."""
         return self._main_chain_residue_sequence
 
     def _build_main_chain(
         self,
-        main_chain_len: int,
-        main_chain_residue_sequence: List[str],
-        side_chain_lens: List[int],
-        side_chain_residue_sequences: List[Optional[str]],
+        main_chain_residue_sequence: str,
+        side_chain_residue_sequences: List[str],
     ) -> Sequence[MainBead]:
         """
         Creates a main chain for a given main chain length and side chain data.
 
         Args:
-            main_chain_len: length of the main chain of a peptide.
-            main_chain_residue_sequence: list of characters that define residues for a main chain.
-            side_chain_lens: list of lengths of all side chains.
-            side_chain_residue_sequences: list of characters that define residues for all side
-                                        beads. None if a side bead does not exist.
+            main_chain_residue_sequence: String of characters that define residues for a main chain.
+            side_chain_residue_sequences: List of characters that define residues for all side
+                                        beads. Empty string if a side bead does not exist.
 
         Returns:
             An instance of a MainChain class.
 
         Raises:
-            InvalidSizeException: when the length of list of side chain lengths provided does not
+            InvalidSizeException: When the length of list of side chain lengths provided does not
```

Comment:
Since you changed it above, maybe do the same here. I am not sure about other occurrences.